src/api/app/models/workflow/step/branch_package_step.rb

```diff
@@ -14,43 +14,53 @@ def initialize(step_instructions:, scm_extractor_payload:, token:)
       end
 
       def allowed_event_and_action?
-        new_pull_request?
+        new_pull_request? || updated_pull_request?
       end
 
       def call
-        return unless valid? && new_pull_request?
+        return unless valid? && allowed_event_and_action?
 
-        branched_package = branch
+        branched_package = find_or_create_branched_package
         return unless branched_package
 
-        add_branch_request_file(package: branched_package)
-
+        add_or_update_branch_request_file(package: branched_package)
         create_or_update_subscriptions(branched_package)
         branched_package
       end
 
-      def source_project
+      def source_project_name
         @step_instructions['source_project']
       end
 
-      def source_package
+      def source_package_name
         @step_instructions['source_package']
       end
 
-      def target_package
+      def target_package_name
         return @step_instructions['target_package'] if @step_instructions['target_package'].present?
 
-        source_package
+        source_package_name
       end
 
-      def target_project
-        "home:#{@token.user.login}:#{source_project}:PR-#{@scm_extractor_payload[:pr_number]}"
+      def target_project_name
+        "home:#{@token.user.login}:#{source_project_name}:PR-#{@scm_extractor_payload[:pr_number]}"
       end
 
       private
 
+      def target_package
+        Package.find_by_project_and_name(target_project_name, target_package_name)
+      end
+
+      def find_or_create_branched_package
+        return target_package if updated_pull_request? && target_package.present?
+
+        branched_package = branch
+        return branched_package if branched_package
```

Comment:
```suggestion
        branch # returns a branched package or nil, so the if is not really needed
```

Reply:
returns a branched package or nil, so the if is not really needed

Reply:
done